app/src/main/java/com/appunite/loudius/network/datasource/AuthDataSource.kt

```diff
@@ -47,16 +45,15 @@ class AuthNetworkDataSource @Inject constructor(
     override suspend fun getAccessToken(
         clientId: String,
         clientSecret: String,
-        code: String,
+        code: String
     ): Result<AccessToken> =
-        apiRequester.safeApiCall { authService.getAccessToken(clientId, clientSecret, code) }
-            .flatMap { response ->
-                if (response.accessToken != null) {
-                    Result.success(response.accessToken)
-                } else {
-                    Result.failure(response.mapErrorToException())
-                }
+        authService.getAccessToken(clientId, clientSecret, code).flatMap { response ->
+            if (response.accessToken != null) {
+                Result.success(response.accessToken)
+            } else {
+                Result.failure(response.mapErrorToException())
             }
+        }
 
```

Comment:
I am still not sure if we realllly need this layer, we could just do it inside AuthService

Reply:
Can you give me an example how to do it in `AuthService`? :) Because in `AuthDataSource` we are retrieving `AccessTokenResponse` which can contain error or access token and then return `AccessToken` (which is alias for string) and I am wondering how to do it in `AuthService` and still have `AccessTokenResponse`.

Reply:
I would just  do all the above inside AuthService, but once again, it i just a suggestion nothing blocking if you really see a value in such a split that we could benefit in the future, please go ahead :P It would mean that the authservice would return either success or a failure like AuthDataSource does